src/main/java/codechicken/nei/recipe/GuiOverlayButton.java

```diff
@@ -151,20 +183,16 @@ public void mouseReleased(int mouseX, int mouseY) {
     public void drawItemPresenceOverlay() {
         final int presenceOverlay = NEIClientConfig.itemPresenceOverlay();
         final boolean highlightPresentItem = NEIClientConfig.isSlotHighlightPresent();
-        final List<PositionedStack> ingredients = this.handler.getIngredientStacks(recipeIndex);
-        final List<Boolean> itemPresenceCache = ingredientsOverlay();
+        final ItemOverlayFormat format = presenceOverlay == 2 ? ItemOverlayFormat.BACKGROUND : ItemOverlayFormat.ICON;
```

Comment:
Instead of hardcoding conversion logic somewhere in implementation, it's better to create static method in `ItemOverlayFormat` enum and hide implementation details:
```java
public static ItemOverlayFormat from(int overlay) {
	return presenceOverlay == 2 ? ItemOverlayFormat.BACKGROUND : ItemOverlayFormat.ICON;
}
```

And then use it here:
```java
final ItemOverlayFormat format = ItemOverlayFormat.from(presenceOverlay);
```

Reply:
done